modules/mauth-signer-akka-http/src/main/scala/com/mdsol/mauth/http/TraceHttpRequest.scala

```diff
@@ -0,0 +1,20 @@
+package com.mdsol.mauth.http
+
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.model.{HttpHeader, HttpRequest}
+import brave.Span
+
+import scala.collection.immutable.Seq
+
+final class TraceHttpRequest(val request: HttpRequest) {
+  def withTraceHeaders[T](childSpan: Span): TraceHttpRequest = {
+    val traceHeaders: Seq[HttpHeader] = Map(
+      TraceHeaders.TRACE_ID_NAME -> Option(childSpan.context().traceId()),
```

Comment:
first two entries are not really optional.

Reply:
Fair enough, although I'd probably create non optional first and then concatenate with optional ones.